homework6/counter.py

```diff
@@ -0,0 +1,45 @@
+"""
+Написать декоратор instances_counter, который применяется к любому классу
+и добавляет ему 2 метода:
+get_created_instances - возвращает количество созданых экземпляров класса
+reset_instances_counter - сбросить счетчик экземпляров,
+возвращает значение до сброса
+Имя декоратора и методов не менять
+
+Ниже пример использования
+"""
+
+
+def instances_counter(cls):
+    """Some code"""
+    class AClass:
+        __instance = 0
+
+        def __init__(self, *args, **kwargs):
+            AClass.__instance += 1
+
+        @classmethod
+        def get_created_instances(cls) -> int:
+            return cls.__instance
+
+        @classmethod
+        def reset_instances_counter(cls) -> int:
+            count = cls.__instance
+            print("before " + str(cls.__instance))
+            AClass.__instance = 0
+            print(cls.__instance)
+            return count
+    return AClass
```

Comment:
if you do this way, you would actually replace the decorated class by AClass and delete all other methods. try doing smth like cls.method_to_change = new_method